### DISCUSSION TITLE
p/452a85351b7f4a31b56cf25eb9d3ba50/

### DISCUSSION THREAD

**Discussion by giscus** [rxliuli/blog#26] [Announcements]
# p/452a85351b7f4a31b56cf25eb9d3ba50/

场景 官方文档  在两年前，vue3 发布之后不久创建了一个 sfc 提案，它允许所有 script 中顶级变量默认绑定到模板，以消除和 react 的开发者体验的差距之一。 大概长这样 12345678910111213141516<script setup>// imported components are also directly usable in templateimpo

https://blog.rxliuli.com/p/452a85351b7f4a31b56cf25eb9d3ba50/

**Comment by qijizh** [comment]
对复杂类型和外部导入类型的局限性是因为ts-language-server不能给sfc-language-server提供足够的信息吗，如果使用jsx编写vue组件是否就能享受完全的类型体验

**Reply by rxliuli** [reply]
单纯是 vue 插件自身的限制，有插件解决
